main/console/console.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2015 - 2016 Square Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/signal"
+	"runtime/pprof"
+	"syscall"
+	"time"
+
+	"github.com/square/metrics/log"
+	"github.com/square/metrics/metric_metadata"
+	"github.com/square/metrics/metric_metadata/cached"
+	"github.com/square/metrics/metric_metadata/cassandra"
+	"github.com/square/metrics/query/command"
+	"github.com/square/metrics/query/parser"
+	"github.com/square/metrics/timeseries"
+
+	"github.com/square/metrics/function/registry"
+	"github.com/square/metrics/main/common"
+	"github.com/square/metrics/timeseries/blueflood"
+	"github.com/square/metrics/util"
+)
+
+func main() {
+	//Adding a signal handler to dump goroutines
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGUSR2)
+
+	go func() {
+		for range sigs {
+			pprof.Lookup("goroutine").WriteTo(os.Stdout, 1)
+		}
+	}()
+
+	config := struct {
+		ConversionRulesPath string           `yaml:"conversion_rules_path"`
+		Cassandra           cassandra.Config `yaml:"cassandra"`
+		Blueflood           blueflood.Config `yaml"blueflood"`
+	}{}
+
+	common.LoadConfigs(&config)
+
+	metadataAPI, err := cassandra.NewMetricMetadataAPI(config.Cassandra)
+	if err != nil {
+		common.ExitWithErrorMessage("Error loading Cassandra API: %s", err.Error())
+		return
+	}
+
+	// TODO: should Blueflood be responsible for this?
```

Comment:
Yeah, it's a weird case because some of Blueflood's config fields (tenant, host, etc.) are filled from YAML, but others have to be filled in afterward (namely the graphite converter).

The current situation is probably fine. I'll remove this comment.